### PR TITLE
feat: segmented checklist progress bar

### PR DESCRIPTION
**Pull request by Shreesh09**
[FR] #3669 

### Feature Preview
Added segments to the checklist progress bar for number of tasks less than or equal to 5 

---

https://github.com/AppFlowy-IO/AppFlowy/assets/114166000/1d0ae319-183a-42fa-b858-c6017f63ab11

#### PR Checklist

- [x] My code adheres to [AppFlowy's Conventions](https://docs.appflowy.io/docs/documentation/software-contributions/conventions)
- [x] I've listed at least one issue that this PR fixes in the description above.
- [ ] I've added a test(s) to validate changes in this PR, or this PR only contains semantic changes.
- [x] All existing tests are passing.

### CODE REVIEW

**Comment by richardshiue** [comment]
I'd much prefer to use a less expensive widget for each of the segment than `LinearPercentIndicator` since each segment will be completely filled

**Comment by Shreesh09** [comment]
Yes that makes a lot of sense, I will implement the changes right away

File: frontend/appflowy_flutter/lib/plugins/database_view/widgets/row/cells/checklist_cell/checklist_progress_bar.dart [hunk 1]
```diff
@@ -18,13 +25,41 @@ class _ChecklistProgressBarState extends State<ChecklistProgressBar> {
...
-            progressColor: Theme.of(context).colorScheme.primary,
-            backgroundColor: AFThemeExtension.of(context).progressBarBGColor,
-            barRadius: const Radius.circular(5),
+          child: Row(
+            children: [
+              if (widget.tasks.isNotEmpty &&
+                  widget.tasks.length <= widget.segmentLimit) ...[
+                for (int i = 0,
+                        j = widget.tasks.where((e) => e.isSelected).length;
+                    i < widget.tasks.length;
```
Comment (richardshiue) [inline frontend/appflowy_flutter/lib/plugins/database_view/widgets/row/cells/checklist_cell/checklist_progress_bar.dart:34]: I think you can calculate what is the number of completed tasks and incompleted tasks, then do a spread operator on a List.generate

File: frontend/appflowy_flutter/lib/plugins/database_view/widgets/row/cells/checklist_cell/checklist_progress_bar.dart [hunk 2]
```diff
@@ -18,13 +25,41 @@ class _ChecklistProgressBarState extends State<ChecklistProgressBar> {
...
+                    i++, j--)
+                  Expanded(
```
Comment (richardshiue) [inline frontend/appflowy_flutter/lib/plugins/database_view/widgets/row/cells/checklist_cell/checklist_progress_bar.dart:36]: I think Flexible would be more suitable here

Comment (Shreesh09) [reply frontend/appflowy_flutter/lib/plugins/database_view/widgets/row/cells/checklist_cell/checklist_progress_bar.dart:36]: Thank you for the input, I will make both the changes

**Review by richardshiue — APPROVED** [review]
LGTM, thanks for the contribution!

**Pull request merged**